src/bp/lang-server/service/index.ts

```diff
@@ -41,10 +62,34 @@ export default class LanguageService {
     this._models = this._getModels()
     const languages = Object.keys(this._models)
 
+    if (languages.length > 0) {
+      const ramInfos = checkEnoughRam(languages.length, this.dim)
+
+      if (ramInfos.prediction + ramInfos.prediction * MARGIN_PRED > ramInfos.free) {
+        this.logger.warn(
+          'Warning, this program can crash silently if it lacks free memory space. Make you sure you have enough RAM available.'
+        )
+        this.logger.warn(
+          `Lang server will take ${ramInfos.prediction}Gb and you have approximately ${ramInfos.free}Gb left on ${ramInfos.total}`
+        )
+      }
+    }
+
     this.logger.info(`Found Languages: ${!languages.length ? 'None' : languages.join(', ')}`)
     await Promise.mapSeries(languages, this._loadModels.bind(this))
 
     this._ready = true
+
+    setInterval(() => {
+      const ramInfos = checkEnoughRam(languages.length, this.dim)
+
+      if (ramInfos.free < ramInfos.total * MARGIN_LOW) {
+        this.logger.warn(
+          `Warning, the language server may silently crash due to a lack of free memory space. (${ramInfos.total -
+            ramInfos.free}/${ramInfos.total})`
+        )
+      }
+    }, 60000)
```

Comment:
use the ms package please

`ms('1m')`

Reply:
needed a comment to finish the review on vscode, useful extension but i still don't get everything